public/js/src/module/records-queue.js

```diff
@@ -27,15 +28,16 @@ let uploadOngoing = false;
  */
 
 function init() {
-    _setUploadIntervals();
-
     // TODO: Add export feature
 
     $exportButton = $('.record-list__button-bar__button.export');
     $uploadButton = $('.record-list__button-bar__button.upload');
     $queueNumber = $('.offline-enabled__queue-length');
 
-    return _updateRecordList();
+    finalRecordPresent = false;
+    uploadOngoing = false;
+
+    return _updateRecordList().then(uploadQueue);
```

Comment:
Because this happens [before the loading indicator is removed](https://github.com/enketo/enketo-express/blob/80e28a24baef4e03df4a9b98509e04cb3e78b78f/public/js/src/module/controller-webform.js#L120-L123), the reassuring `__recordNames__ were successfully submitted` banner doesn't appear if a user has previously queued submissions when the app initializes.

One example workflow:
1. Have queued submissions (and see a number > 0 in the top, left corner);
1. Become unauthenticated somehow (logging out in a new tab is an artificial way, but maybe you use forms belonging to multiple accounts or your authentication times out);
1. Get prompted to log in (assuming OpenRosa auth which redirects to a separate page);
1. Log in and get redirected back to the app;
1. See the loading indicator;
1. See that the "Records Queued" number in the top-left has immediately gone to zero.
    * Of course, this is because the submissions were uploaded fast :tada: but it may unsettle people who are used to seeing some kind of confirmation of transmittal.

I can work around this by moving the initial `uploadQueue()` call into `controller-webform.js`, after the loading indicator is removed (https://github.com/jnm/enketo-express/commit/b13b96d74046fdcc9e8126780411d0170028657f), and it works nicely: the reassuring banner appears. Unfortunately, taking the initial upload out of `records-queue.js` breaks your valuable "uploads submissions which were previously queued before initializing the current session" test, and I'm not sure how to replace it without extensive work.